libraries/ts-command-line/src/parameters/CommandLineChoiceListParameter.ts

```diff
@@ -43,17 +43,20 @@ export class CommandLineChoiceListParameter extends CommandLineParameter {
    */
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   public _setValue(data: any): void {
-    // If argparse passed us a value, confirm it is valid
+    // If argparse passed us a value, confirm it is valid. Note that argparse
+    // may pass an array of arrays of args, so we flatten it before validating
+    // each entry.
     if (data !== null && data !== undefined) {
       if (!Array.isArray(data)) {
         this.reportInvalidData(data);
       }
-      for (const arrayItem of data) {
-        if (typeof arrayItem !== 'string') {
+      const values: unknown[] = data.reduce((list, item) => list.concat(item), []);
+      for (const item of values) {
+        if (typeof item !== 'string') {
           this.reportInvalidData(data);
         }
       }
```

Comment:
```suggestion
      const values: unknown[] = [];
      for (const items of data) {
        for (const subItem of items) {
          values.push(item);
          if (typeof item !== 'string') {
            this.reportInvalidData(data);
          }
        }
      }
```

Eliminates the double-loop.